### DISCUSSION TITLE
How to build a release (not development) version?

### DISCUSSION THREAD

**Discussion by 1067088037** [Budibase/budibase#6877] [Contributing]
Hi, I have read the _CONTRIBUTING.md_ and known how to run a development version.
I wonder how to package it as a release version (may be a docker image).
Should I continue to use a script in YARN or is there another way?

I would be grateful if you could answer.

**Comment by mike12345567** [comment]
Hi @1067088037 - this process isn't admittedly very well documented, we should fix that.

From the top level of the Budibase repo if you run `yarn build:docker:pre` then `yarn build:docker` you should have images produced at the end.

Currently we have it setup so that it will automatically attempt to push the image when it is finished being built (see the `./release-to-docker-hub.sh` script being ran at the end of the command) so you will need to be logged into docker hub via the Docker CLI for this to work.

You can also build our new single image (the previous method generates the images required for our docker compose deployment) using the command `yarn build:docker:single`.

Its worth noting that this likely will only work on Unix based systems, we haven't tested the build process on Windows, it is likely that you will need to use WSL instead if working in a Windows environment.

**Comment by chripo** [comment]
dosen't work. the task fails with
```
 => ERROR [build  8/15] RUN yarn install --production=true                                                                      3.1s
------
 > [build  8/15] RUN yarn install --production=true:
0.424 yarn install v1.22.19
0.633 [1/4] Resolving packages...
3.039 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
3.040 error Couldn't find any versions for "@budibase/backend-core" that matches "0.0.0"
------
Dockerfile:15
--------------------
  13 |     ADD packages/server .
  14 |     COPY yarn.lock .
  15 | >>> RUN yarn install --production=true
  16 |     RUN /cleanup.sh
  17 |
--------------------
ERROR: failed to solve: process "/bin/sh -c yarn install --production=true" did not complete successfully: exit code: 1
```

**Comment by mikeluigijean** [comment]
I have the same issue. The task fails for me also with the same error as @chripo
